src/coreclr/vm/dllimportcallback.h

```diff
@@ -227,7 +244,8 @@ class UMEntryThunkData
         if (m_pObjectHandle == NULL && m_pManagedTarget == (TADDR)0)
             m_pManagedTarget = m_pMD->GetMultiCallableAddrOfCode();
 
-        m_pUMEntryThunk->SetTargetUnconditional(m_pUMThunkMarshInfo->GetExecStubEntryPoint());
+        // FIXME: Do we need to use atomics here?
+        m_pCachedCallTarget = m_pUMThunkMarshInfo->GetExecStubEntryPoint();
```

Comment:
```suggestion
        PCODE pILStub = m_pUMThunkMarshInfo->GetExecStubEntryPoint();

#ifdef FEATURE_INTERPRETER
        // For interpreted stubs we need to ensure that TheUMEntryPrestubWorker runs for every
        // unmanaged-to-managed invocation in order to populate the TLS variable every time.
        auto stubKind = RangeSectionStubManager::GetStubKind(pILStub);
        if (stubKind == STUB_CODE_BLOCK_STUBPRECODE)
        {
            StubPrecode* pPrecode = Precode::GetPrecodeFromEntryPoint(pILStub)->AsStubPrecode();
            if (pPrecode->GetType() == PRECODE_INTERPRETER)
            {
                m_pIntepretedCallTarget = pILStub;
                pILStub = NULL;
            }
        }

        if (pILStub != NULL)
#endif // FEATURE_INTERPRETER
        {
            m_pUMEntryThunk->SetTargetUnconditional(pILStub);
        }
```
I think we should be able to do all this here. Also, can we check for InterpreterPrecode directly? We use `auto stubKind = RangeSectionStubManager::GetStubKind(` pattern in number of other places to trigger interpreter specific logic.

Reply:
The InterpreterJitManager doesn't have the GetStubCodeBlockKind virtual method really implemented yet. It currently just returns STUB_CODE_BLOCK_UNKNOWN, so we will need to implement it.

Reply:
Ah, we won't need it here as the pILStub is obviously not managed by the interpreter manager.